From 93d8de3a5109d473c7112b683cf7f44ef0ded840 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 2 Apr 2024 18:27:36 +0200
Subject: [PATCH 0794/1695] plat-stm32mp2: add HPDMA Address Remapping SYSCFG
 support

Adds support for the HPDMA Address Remapping configuration that is present
in SYSCFG.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 71e4c718f9b684bcaa7388de71c2f6a5e36dd574 plat-stm32mp2: add HPDMA Address Remapping SYSCFG support]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: Ia3c3eaae8acaefd987b52a5074dea12733c076f9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372996
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/stm32_sysconf.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/stm32_sysconf.h b/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
index 3e86cd779..b16d56d69 100644
--- a/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
+++ b/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
@@ -27,6 +27,7 @@ enum syscon_banks {
  */
 #define SYSCFG_VDERAMCR		SYSCON_ID(SYSCON_SYSCFG, 0x1800U)
 #define SYSCFG_POTTAMPRSTCR	SYSCON_ID(SYSCON_SYSCFG, 0x1804U)
+#define SYSCFG_HPDMAARCR	SYSCON_ID(SYSCON_SYSCFG, 0x2050U)
 
 /*
  * SYSCFG_VDERAMCR register offsets
@@ -34,6 +35,13 @@ enum syscon_banks {
 #define VDERAMCR_VDERAM_EN		BIT(0)
 #define VDERAMCR_MASK			BIT(0)
 
+/*
+ * SYSCFG HPDMA address remapping control register (SYSCFG_HPDMAARCR)
+ */
+#define SYSCFG_HPDMAARCR_HPDMA1AREN	BIT(0)
+#define SYSCFG_HPDMAARCR_HPDMA2AREN	BIT(1)
+#define SYSCFG_HPDMAARCR_HPDMA3AREN	BIT(2)
+
 /*
  * A35SSC register offsets (base relative)
  */
-- 
2.47.2

